extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/client/config/ProxyConfig.java

```diff
@@ -3,41 +3,38 @@
 import java.util.Optional;
 
 import io.quarkus.runtime.annotations.ConfigGroup;
-import io.quarkus.runtime.annotations.ConfigItem;
+import io.smallrye.config.WithDefault;
 import io.vertx.core.net.ProxyType;
 
 @ConfigGroup
-public class ProxyConfig {
+public interface ProxyConfig {
 
     /**
      * Set proxy username.
      */
-    @ConfigItem
-    public Optional<String> username;
+    Optional<String> username();
 
     /**
      * Set proxy password.
      */
-    @ConfigItem
-    public Optional<String> password;
+    Optional<String> password();
 
     /**
      * Set proxy port. Defaults to 3128.
      */
-    @ConfigItem(defaultValue = "3128")
-    public int port;
+    @WithDefault("3128")
+    int port();
 
     /**
      * Set proxy host.
      */
-    @ConfigItem
-    public String host;
+    Optional<String> host();
```

Comment:
@cescoffier this is an important change. Before this change, you would have proxy enabled as long as you defined one of the property in `ProxyConfig` whatever it was.
Due to a change on how Optional groups are handled, I had to change it to the following behavior: the proxy is enabled if an `host` is defined. I think this is an acceptable trade-off.

Reply:
@cescoffier just wanted to make sure you have seen ^ as I hope this PR will be ready really soon.